### DISCUSSION TITLE
popen: pipe the buffer into an arbitrary command

### DISCUSSION THREAD

**Discussion by SuperDuperDeou** [ghostty-org/ghostty#3199] [Feature Requests, Ideas]
An action, or a family of actions, that pipe the content of the screen into an arbitrary command, and maybe even paste the output of an arbitrary command.  In the config it would look something like this:
```
# Pipe the visible content of the focused window
keybind = atl+p>q:popen_window:foo_1.sh

# Pipe the scroolback buffer of the focused window
keybind = atl+p>w:popen_scrollback:foo_2.sh

# Pipe the output of the last command
keybind = atl+p>e:popen_output:foo_3.sh

# Pipe the last command
keybind = atl+p>r:popen_command:foo_4.sh

# Pipe a newline-separated list of the URLs in all scrollback buffers
keybind = atl+p>t:popen_url:foo_5.sh

# Paste the output of foo_6.sh
keybind = atl+p>y:popen_paste:foo_6.sh
```

This feature (and tab support) is why I use Kitty over Alacritty.

Edit:
It might look similar to one of the points of #3197 (which came out while I was writing this, lol), but spawning a command inside a Ghostty window is fundamentally different from piping data into a command that might not even have a window at all.

**Comment by msharran** [comment]
Cross posting a relevant comment

https://github.com/ghostty-org/ghostty/issues/189#issuecomment-2564638258
